### PR TITLE
EP-318: Run js test with coverage for Codacy

### PR DESCRIPTION
**Pull request by nilsguillermin**
## JIRA ticket

https://tetrascience.atlassian.net/browse/EP-318

## Description

- Generate coverage when running js tests for Codacy
- Pass in NPM secrets to publish.yml and publish_tag.yml workflows
- Add NPM configuration step to publish_tag.yml workflow

## Supporting test data

This was tested using this PR: https://github.com/tetrascience/ts-task-script-nils-test-js-task-script/pull/1

## Automated testing

- [ ] Unit tests
- [ ] Integration tests
- [ ] API tests
- [ ] End-To-End tests
- [ ] N/A (please clarify)

### CODE REVIEW

File: .github/workflows/javascript_code_quality.yml [hunk 1]
```diff
@@ -90,8 +90,8 @@ jobs:
             echo "::error ::Could not determine yarn version"
           fi
 
-      - name: yarn test
-        run: yarn test
+      - name: yarn test (with coverage)
+        run: yarn test --coverage
```
Comment (aslotte) [inline .github/workflows/javascript_code_quality.yml:94]: I don't think this change is needed. We should instead either assume that `yarn test` produces coverage (my preference) or have a `yarn test:coverage` command

File: .github/workflows/publish.yml [hunk 2]
```diff
@@ -35,6 +35,11 @@ on:
         required: false
       JFROG_ARTIFACTORY_URL:
         required: false
+      JFROG_ARTIFACTORY_NPM_VIRTUAL_URL:
```
Comment (aslotte) [inline .github/workflows/publish.yml:38]: Where are these being used?

File: .github/workflows/publish_tag.yml [hunk 3]
```diff
@@ -298,6 +302,29 @@ jobs:
             echo "No lock file found. Dependency manager not installed."
           fi
 
+      - name: Configure JFrog Credentials (Node)
+        if: inputs.type == 'task-script'
+        env:
+          JFROG_ARTIFACTORY_NPM_VIRTUAL_URL: ${{ secrets.JFROG_ARTIFACTORY_NPM_VIRTUAL_URL }}
+          JFROG_ARTIFACTORY_READ_NPM_AUTH: ${{ secrets.JFROG_ARTIFACTORY_READ_NPM_AUTH }}
+        working-directory: artifact_code
```
Comment (aslotte) [inline .github/workflows/publish_tag.yml:310]: Should this also be done in the `publish.yml`?

Comment (nilsguillermin) [reply .github/workflows/publish_tag.yml:310]: It already is (thus answering your previous question).

File: .github/workflows/publish.yml (hunk 2 above)
Comment (nilsguillermin) [reply .github/workflows/publish.yml:38]: In configuring NPM credentials so the CI can download `ts-lib-node-sdk` which is what builds the JS artifacts. See: https://github.com/tetrascience/ts-data-insights-ci-cd-common/blob/EP-318-fix-js-linting/.github/workflows/publish.yml#L239

File: .github/workflows/javascript_code_quality.yml (hunk 1 above)
Comment (nilsguillermin) [reply .github/workflows/javascript_code_quality.yml:94]: Going with `yarn test` for simplicity, though I'm not sure everyone wants to generate coverage every time they run tests.

**Review by aslotte — APPROVED** [review]

**Pull request merged**